_posts/2022-11-11-software-factory.markdown

```diff
@@ -0,0 +1,127 @@
+
+**Consistent**, because we've developed our own internal set of functions to be used with our build server and writing also build pipelines themselves as a Blue Print for many projects. In this way, users of the build server don't need to spend a lot of time writing their own pipeline, but can use ready-to-go solutions that fit their projects. This accelerates the development process, shifting the focus on solving business problems, not infrastructure ones.
+
+## Migration
+
+We moved gematik onPrem Build- and Deployment System to fully cloud based scalable Software Factory in several steps with minimal work and workflow disruption on product teams at the last years.
+
+First step was the renew of our old windows based Jenkins System with static Agents, to a completely new Linux based Jenkins Master and declarative Pipeline definitions checkin with the project sourcecode. The active Projects has a long time to migrate the old Jenkins-Jobs to the new declarative Pipelines and old Projects was archived.
+
+The number of builds increase in the following month because we start with attentional Feature-Branch and Merge-Request Builds. The virtual Machine based static Jenkins Agents was not enough and we start with dynamic Docker based Agents. This new Agent concept help us to fulfill the requirement of reproducible builds on defined environment without several builds on the same Agent, every build starts a new clean Agent.
+The types of our Jenkins Agents various from Maven-Java to Angular Agents and result in nine Agent-types.
+
+The number of teams and projects increased from month to month and the available hardware for the agents became increasingly scarce, which led to ever longer build times. The next step we took was integrating resources from the google cloud and moving Jenkins agents to a Kubernetes cluster. The number of parallel builds increased significantly, but network communication with the onPrem systems (Git, Nexus, Sonarqube, ...) increased significantly and slowed down the builds.
+
+The last step we took is the migration of all build system components into a google Kubernetes cluster and name it 'Software Factory'.
+
+## What is 'Software Factory'?
+
+At the time of writing, the Software Factory is a composition of following tools:
+
+* GitLab as central CVS server
+* Jenkins and custom Build-Agents for the different projects
+* SonaType Nexus, for hosting our build artifacts and Helm Charts
+* SonarQube, enforcing Quality Gates on code
+* Google Container Registry to store Docker Images built in the process
+* Different sets of scanners for analyzing code and built artifacts
+* Kubernetes and some on Premise Servers for hosting applications
+* Monitoring Solutions
+
+![Sketch of the matrix organization @gematik]({{site.baseurl}}/assets/img/221111-swfactory/Software-Factory.webp)
```

Comment:
is the file missing? image is not shown...

Reply:
through move from external repository, images went missing. added back